egg/core/interaction.py

```diff
@@ -131,13 +135,33 @@ def _check_cat(lst):
                 )
             return torch.cat(lst, dim=0)
 
+        def _aux_input_cat(lst):
+            """
+            Transform aux inputs from list of dict to dict of list extending the original key
+            """
+            # filter out None and return {} if empty
+            lst = [x for x in lst if x is not None]
+            if len(lst) == 0:
+                return {}
+
+            # build final dict
+            cat = {k: [] for k in lst[0].keys()}
+            # populate
+            for elem in lst:
+                for k, v in elem.items():
+                    cat[k] += v
+            
+            return cat
+
         assert interactions, "list must not be empty"
         assert all(len(x.aux) == len(interactions[0].aux) for x in interactions)
 
         aux = {}
         for k in interactions[0].aux:
             aux[k] = _check_cat([x.aux[k] for x in interactions])
 
+        aux_input = _aux_input_cat([x.aux_input for x in interactions])
```

Comment:
should this be inside check_cat where we do all the checks instead of separating it here?